pkg/moov/errors.go

```diff
@@ -0,0 +1,46 @@
+package moov
+
+import "errors"
+
+func ErrorAsCallResponse(err error) *CallResponse {
+	return errorAsA[CallResponse](err)
+}
+
+func ErrorAsHttpCallResponse(err error) *HttpCallResponse {
+	return errorAsA[HttpCallResponse](err)
+}
+
+func errorAsA[A interface{}](err error) *A {
+	t := new(A)
+	if errors.As(err, t) {
+		return t
+	}
+	return nil
+}
```

Comment:
Why doesn't `errors.As` work?

Reply:
It does, this is a helper to just return the object rather than having to specify all of these lines to do it. Look at above for how to use the helper.